tests/functional/ext/function_return_not_assigned/function_return_not_assigned.py

```diff
@@ -0,0 +1,56 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring, missing-class-docstring, expression-not-assigned, invalid-name
+from dataclasses import dataclass, replace
+
+
+def func_that_returns_something():
+    return 1
+
+
+func_that_returns_something()  # [function-return-not-assigned]
+
+_ = func_that_returns_something()
+
+if func_that_returns_something():
+    pass
+
+
+def func_that_returns_none():
+    return None
+
+
+def func_with_no_explicit_return():
+    print("I am doing something")
+
+
+func_that_returns_none()
+func_with_no_explicit_return()
+
+some_var = ""
+# next line should probably raise?
+func_that_returns_something() if some_var else func_that_returns_none()
```

Comment:
parent of all three `func_that_returns_something` `some_var` and `func_that_returns_none` is IfExpr. If `some_var` would be function call, I would consider it to be used while `func_that_returns_something` `func_that_returns_none` not. `func_that_returns_something` should raise `function-return-not-assigned`